project-api-tests/src/test/java/org/ballerina/projectapi/BalToolTest.java

```diff
@@ -0,0 +1,1008 @@
+        }
+
+        // Execute the tool
+        Pair<String, String> outputs = executeHelpFlagOfTool(
+                toolId, toolEnvironments.get(ToolSubCommand.PULL).envVariables);
+        Assert.assertEquals("", outputs.getLeft());
+        Assert.assertEquals(outputs.getRight(), readExpectedCmdOutsAsString("tool-execute-specific-help-1.1.0.txt"));
+    }
+
+    @Test(description = "Pull tool with a specific version",
+            dependsOnMethods = {"testPullToolAgainWithoutVersion"}, groups = {"pull"})
+    public void testPullToolWithASpecificVersion() throws IOException, InterruptedException {
+        String toolIdAndVersion = toolId + ":" + specificVersion;
+        Process cmdExec = executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", toolIdAndVersion)),
+                toolEnvironments.get(ToolSubCommand.PULL).envVariables);
+        String cmdErrors = getString(cmdExec.getErrorStream());
+        if (!cmdErrors.isEmpty()) {
+            Assert.fail(OUTPUT_CONTAIN_ERRORS + cmdErrors);
+        }
+
+        // Validate the command output
+        String cmdOutput = getString(cmdExec.getInputStream());
+        String expectedOutput = readExpectedCmdOutsAsString("tool-pull-with-specific-version.txt");
+        if (!cmdOutput.contains(expectedOutput)) {
+            Assert.fail(OUTPUT_NOT_CONTAINS_EXP_MSG + expectedOutput + "\nactual output:" + cmdOutput);
+        }
+
+        // Check for tool availability in local env
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.PULL).balToolsTomlPath);
+        Optional<BalToolsManifest.Tool> toolOpt = BalToolsManifestBuilder.from(balToolsToml).build()
+                .getActiveTool(toolId);
+        if (toolOpt.isEmpty()) {
+            Assert.fail("Tool " + toolId + " is not available in bal-tools.toml");
+        }
+        BalToolsManifest.Tool tool = toolOpt.get();
+        Assert.assertEquals(tool.id(), toolId);
+        Assert.assertEquals(tool.version(), specificVersion);
+        Assert.assertEquals(tool.org(), orgName);
+        Assert.assertEquals(tool.name(), packageName);
+
+        Path toolVersionCachePath = toolEnvironments.get(ToolSubCommand.PULL).centralCachePath
+                .resolve(Path.of(tool.org(), tool.name(), tool.version()));
+        if (!Files.exists(toolVersionCachePath)) {
+            Assert.fail("Tool " + toolId + ":" + specificVersion + " is not available in local cache");
+        }
+
+        // Execute the tool
+        Pair<String, String> outputs = executeHelpFlagOfTool(
+                toolId, toolEnvironments.get(ToolSubCommand.PULL).envVariables);
+        Assert.assertEquals("", outputs.getLeft());
+        Assert.assertEquals(outputs.getRight(), readExpectedCmdOutsAsString("tool-execute-specific-help-1.0.0.txt"));
+    }
+
+    @Test(description = "Pull tool again with a specific version",
```

Comment:
Why are we pulling again?

Reply:
When we pull a specific version for the first time, it gets pulled and the version is set as the active version. When we pull the same version again, it simply notifies the user that this is already pulled and is the current version. I have added two tests to test this difference in behaviour.